fast_barcode_scanner/ios/Classes/BarcodeReader.swift

```diff
@@ -134,32 +135,44 @@ class BarcodeReader: NSObject {
 		}) else {
 			throw ReaderError.cameraNotSuitable(arguments.resolution, arguments.framerate)
 		}
-
-		try! captureDevice.lockForConfiguration()
-		captureDevice.activeFormat = optimalFormat
-		captureDevice.activeVideoMinFrameDuration = optimalFormat.videoSupportedFrameRateRanges.first!.minFrameDuration
-		captureDevice.activeVideoMaxFrameDuration = optimalFormat.videoSupportedFrameRateRanges.first!.minFrameDuration
-		captureDevice.unlockForConfiguration()
-
+        
+        do {
+            try captureDevice.lockForConfiguration()
+            captureDevice.activeFormat = optimalFormat
+            captureDevice.activeVideoMinFrameDuration = optimalFormat.videoSupportedFrameRateRanges.first!.minFrameDuration
+            captureDevice.activeVideoMaxFrameDuration = optimalFormat.videoSupportedFrameRateRanges.first!.minFrameDuration
+            captureDevice.unlockForConfiguration()
+        } catch {
+            throw ReaderError.configurationLockError(error)
+        }
+        
 		previewSize = CMVideoFormatDescriptionGetDimensions(captureDevice.activeFormat.formatDescription)
 	}
 
-	func start(fromPause: Bool) {
+	func start(fromPause: Bool) throws {
+        guard captureDevice != nil else { return }
+        
 		captureSession.startRunning()
 
 		if !fromPause {
 			self.textureId = textureRegistry.register(self)
 		}
 
 		if (torchActiveBeforeStop) {
-			try! captureDevice.lockForConfiguration()
-			captureDevice.torchMode = .on
-			captureDevice.unlockForConfiguration()
-			torchActiveBeforeStop = false
+            do {
+                try captureDevice.lockForConfiguration()
+                captureDevice.torchMode = .on
+                captureDevice.unlockForConfiguration()
+                torchActiveBeforeStop = false
```

Comment:
Is this why the torch turns off every time the capture pauses? I was trying to capture in batch mode, but pausing and then resuming and noticed that the torch did not stay on. I guess this could be handled on the front end code though.

Reply:
AVFoundation turns the torch off if you stop the capture session. The code you referenced should reactive the torch if it was active when the capture session was stopped.

What is your configuration?